sdk/core/azure-core-tracing-opentelemetry-samples/src/samples/java/com/azure/core/tracing/opentelemetry/samples/TracingJavaDocCodeSnippets.java

```diff
@@ -0,0 +1,166 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.core.tracing.opentelemetry.samples;
+
+import com.azure.core.http.rest.Response;
+import com.azure.core.tracing.opentelemetry.OpenTelemetryTracingOptions;
+import com.azure.core.util.ClientOptions;
+import com.azure.core.util.Context;
+import com.azure.core.util.TracingOptions;
+import com.azure.core.util.tracing.TracerProvider;
+import io.opentelemetry.api.GlobalOpenTelemetry;
+import io.opentelemetry.api.trace.Span;
+import io.opentelemetry.api.trace.Tracer;
+import io.opentelemetry.context.Scope;
+import io.opentelemetry.exporter.logging.LoggingSpanExporter;
+import io.opentelemetry.sdk.trace.SdkTracerProvider;
+import io.opentelemetry.sdk.trace.export.SimpleSpanProcessor;
+
+import static com.azure.core.util.tracing.Tracer.PARENT_TRACE_CONTEXT_KEY;
+
+/**
+ * Contains code snippets showing how to configure Azure Client libraries to use OpenTelemetry Tracing plugin.
+ */
+public class TracingJavaDocCodeSnippets {
+
+    @SuppressWarnings("try")
+    public void sampleDefaultSdkConfiguration() {
+        // BEGIN: com.azure.core.util.tracing.TracingOptions#default
+
+        // configure OpenTelemetry SDK using io.opentelemetry:opentelemetry-sdk-extension-autoconfigure
+        // AutoConfiguredOpenTelemetrySdk.initialize();
+
+        // configure Azure Client, no metric configuration needed
+        AzureClient sampleClient = new AzureClientBuilder()
+            .endpoint("https://my-client.azure.com")
+            .build();
+
+        Span span = GlobalOpenTelemetry.getTracer("azure-core-samples")
+            .spanBuilder("doWork")
+            .startSpan();
+
+        try (Scope scope = span.makeCurrent()) {
+            // do some work
+
+            // Current context flows to underlying Azure client explicitly
+            // and is used as a parent on span-to-be-created for sampleClient.methodCall.
+            String response = sampleClient.methodCall("get items");
+            // do more work
+        }
+
+        span.end();
+
+        // END: com.azure.core.util.tracing.TracingOptions#default
+    }
+
+    public void customProviderSdkConfiguration() {
+        // BEGIN: com.azure.core.tracing.TracingOptions#custom
+
+        // configure OpenTelemetry SDK explicitly per https://opentelemetry.io/docs/instrumentation/java/manual/
+        SdkTracerProvider tracerProvider = SdkTracerProvider.builder()
+            .addSpanProcessor(SimpleSpanProcessor.create(LoggingSpanExporter.create()))
+            .build();
+
+        // Pass OpenTelemetry tracerProvider to TracingOptions.
+        TracingOptions customTracingOptions = new OpenTelemetryTracingOptions()
+            .setProvider(tracerProvider);
```

Comment:
good point! Since we have separate `TracingOptions` (in this change) and `MetricsOptions` (already released), I prefer to take `TracerProvider` and `MeterProvider`.  

Also, `ContextPropagators` are tricky - we don't want users to control propagator used against Azure services (makes no sense to use B3 against storage), so if we take whole `OpenTelemetry`, we'd ignore propagators anyway.